chapter_04_uow.asciidoc

```diff
@@ -294,8 +305,13 @@ def add_batch(ref: str, sku: str, qty: int, eta: Optional[date], start_uow):
 ----
 ====
 
-This is a judgement call, but we tend to prefer requiring the explicit commit.
-"explicit is better than implicit", and all that.
+This is a judgement call, but we tend to prefer requiring the explicit commit
+so that we have to choose when to flush state.
+
+Although it's an extra line of code this makes the software safe-by-default. The 
+default behaviour is to _not change anything_. In turn, that makes our code
+easier to reason about since any errors in our service function will leave the
```

Comment:
if we always rollback on exception, surely we'll achieve the same thing?

Reply:
No, because that way you'll commit by default. It's literally the opposite.

Reply:
sure but your sentence says "any errors in our service function will leave the system untouched"

Reply:
maybe it's an errors/exceptions misunderstanding?

Reply:
This is haunting me now. I'm not sure how to phrase this. I think it's easier to reason about because there's only one code path that leads to changes in the system: total success. Any other code path, any exception, any early exit from the uow's scope, leads to a safe state.

Reply:
that's a pretty good sentence tbh